packages/terra-section-header/src/SectionHeader.jsx

```diff
@@ -153,17 +153,28 @@ class SectionHeader extends React.Component {
     /* eslint-disable jsx-a11y/click-events-have-key-events */
     /* eslint-disable jsx-a11y/no-static-element-interactions */
 
-    const buttonAttributes = (onClick) ? { role: 'button', 'aria-expanded': isOpen, 'aria-label': headerText } : undefined;
+    // Function to use Arrange component
+    const ArrangeComponent = () => (
+      <Arrange
+        fitStart={onClick && accordionIcon}
+        fill={<span aria-hidden={(onClick !== undefined)} className={cx('title', 'title-block')}>{headerText}</span>}
+        className={cx('title-arrange')}
+      />
+    );
+
+    const buttonAttributes = (onClick) ? { 'aria-expanded': isOpen, 'aria-label': headerText } : undefined;
 
     return (
       <Element {...attributes} className={sectionHeaderClassNames} aria-label={!onClick ? headerText : undefined}>
-        <div {...buttonAttributes} tabIndex="-1" className={cx('arrange-wrapper')}>
-          <Arrange
-            fitStart={onClick && accordionIcon}
-            fill={<span aria-hidden={(onClick !== undefined)} className={cx('title')}>{headerText}</span>}
-            className={cx('title-arrange')}
-          />
-        </div>
+        {onClick && accordionIcon ? (
+          <button {...buttonAttributes} type="button" tabIndex="-1" className={cx('arrange-wrapper toggle-button')}>
+            <ArrangeComponent />
+          </button>
+        ) : (
+          <span tabIndex="-1" className={cx('arrange-wrapper')}>
+            <ArrangeComponent />
+          </span>
```

Comment:
Both of these seem very similar so instead of defining these twice, we can do something like:

```js
const ArrangeWrapperTag = onClick ? "button" : "span" ;
const type = onClick ? "button" : null;
// ... and so on for the other attributes

<ArrangeWrapperTag  (attributes) >
  <ArrangeComponent />
</ArrangeWrapperTag


```

Reply:
Updated.